static/js/filter.js

```diff
@@ -1,430 +1,176 @@
-    function updateAliasLabel(event){
-        const alias = event.target.closest(".js-alias");
-        const prevLabel = event.target.dataset.label;
-        const prevLabelLowercased = prevLabel.toString().toLowerCase();
-        const newLabel = event.target.value;
-        const newLabelLowercased = newLabel.toString().toLowerCase();
-        const AliasWithLabelsArrayIndex = aliasesWithLabelsCollection.indexOf(alias);
-        const labelArrayIndex = filterAliasLabels.indexOf(prevLabelLowercased);
+        });
 
+        filterLabelVisibleCases.textContent = aliasContainers.filter(aliasContainer => !aliasContainer.classList.contains("is-hidden")).length;
+        filterLabelTotalCases.textContent = aliasContainers.length;
 
-        // Case: User did not enter any label, nor was one previously set
-        if (prevLabel === "" && newLabel === "") {
-            return;
-        }
-
-        // Case: User deleted/cleared an existing label
-        if (newLabel === "") {
-            filterAliasLabels.splice(labelArrayIndex, 1);
-            aliasesWithLabelsCollection.splice(AliasWithLabelsArrayIndex, 1);
-            return;
+        if (activeSearchFilter.length > 0) {
+            filterSearchInput.classList.add("is-filtered");
+            filterContainer.classList.add("is-filtered-by-search");
+        } else {
+            filterSearchInput.classList.remove("is-filtered");
+            filterContainer.classList.remove("is-filtered-by-search");
         }
-
-        // Case: User updated an existing label to a new string
-        if (labelArrayIndex > -1) {
-            filterAliasLabels.splice(labelArrayIndex, 1, newLabelLowercased);
-            return;
+        if (activeCategoryFilters.length > 0) {
+            filterContainer.classList.add("is-filtered-by-category");
+        } else {
+            filterContainer.classList.remove("is-filtered-by-category");
         }
-
-        // Case: User created a label for an alias
-        aliasesWithLabelsCollection.push(alias);
-        filterAliasLabels.push(newLabelLowercased);
-
     }
 
-    function isAddOnDetected() {
-        const addNotes = document.querySelector(".additional-notes");
-        if (!addNotes) return false;
-        return (addNotes.offsetWidth > 0 && addNotes.offsetHeight > 0);
+    function setSearchFilter(query) {
+        activeSearchFilter = query;
+
+        applyFilters();
     }
 
-    function buildSearchQueryArrays() {
-        const addOnDetected = isAddOnDetected();
-        
-        // Build two arrays, one for case IDs and one for case title text. 
-        const isCategoryFilterActive = (filterContainer.classList.contains("is-filtered-by-category"));
+    filterSearchInput.addEventListener("input", (event) => {
+        setSearchFilter(event.target.value);
+    });
 
-        let availableAliasesForSearchFilter = aliases;
+    // TODO: Add esc key listener
+    filterResetButton.addEventListener("click", (event) => {
```

Comment:
nit: unused event arg
```suggestion
    filterResetButton.addEventListener("click", () => {
```

Reply:
Fixed in 19bf72a48b16d8c1408f2229478cb62040bbe1dc.